ompi/communicator/comm.c

```diff
@@ -941,26 +941,34 @@ int ompi_comm_split_type (ompi_communicator_t *comm, int split_type, int key,
     global_orig_split_type = tmp[0];
     global_split_type = tmp[4];
 
-    if (tmp[0] != -tmp[1] || inter) {
+    if (tmp[0] != -tmp[1] || tmp[4] != -tmp[5] || inter) {
         /* at least one rank supplied a different split type check if our split_type is ok */
-        ok = (MPI_UNDEFINED == orig_split_type) || global_orig_split_type == orig_split_type;
+        ok[0] = (MPI_UNDEFINED == orig_split_type) || global_orig_split_type == orig_split_type;
+        ok[1] = (MPI_UNDEFINED == orig_split_type) || global_split_type == split_type;
 
-        rc = comm->c_coll->coll_allreduce (MPI_IN_PLACE, &ok, 1, MPI_INT, MPI_MIN, comm,
+        rc = comm->c_coll->coll_allreduce (MPI_IN_PLACE, &ok, 2, MPI_INT, MPI_MIN, comm,
                                           comm->c_coll->coll_allreduce_module);
         if (OPAL_UNLIKELY(OMPI_SUCCESS != rc)) {
             return rc;
         }
 
         if (inter) {
             /* need an extra allreduce to ensure that all ranks have the same result */
-            rc = comm->c_coll->coll_allreduce (MPI_IN_PLACE, &ok, 1, MPI_INT, MPI_MIN, comm,
+            rc = comm->c_coll->coll_allreduce (MPI_IN_PLACE, &ok, 2, MPI_INT, MPI_MIN, comm,
                                               comm->c_coll->coll_allreduce_module);
             if (OPAL_UNLIKELY(OMPI_SUCCESS != rc)) {
                 return rc;
             }
         }
 
-        if (OPAL_UNLIKELY(!ok)) {
+        if (OPAL_UNLIKELY(!ok[0] || !ok[1])) {
+            if (0 == ompi_comm_rank(comm)) {
+                if (!ok[1]) {
+                    opal_output(0, "Error: Mismatched info values for MPI_COMM_TYPE_HW_GUIDED");
```

Comment:
increase the user-friendliness by using opal_show_help here.

Reply:
I made the change. Let me know what you think.